### PR TITLE
Upgrade CODAL with audio pipeline

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Upgrades CODAL and switches out the SOUND_MIRROR_EXTENSION for the new virtual pin in pinAnalogSetPitch. This should get interoperability between existing music blocks and the new playSound once the extension is also updated.

This version will also enable the v2 extension to drop a lot of audio code in favour of simple calls to uBit.audio.

- [x] To set volume on the mixer via new codal-only API I've used C++ which breaks the simulator. I'd love input on the best way to address this.

- [x] Missing a tag for microbit-v2-samples so that's using `master` as the `gittag`. We've asked Joe to tag it.

- Volume needs some more discussion but perhaps that can be a follow-up PR. There's a difference between normal and virtual pin pin volumes and I think the mixer volume doesn't use the same log scale that pxt pitch volume does.

CC @jaustin

### CODE REVIEW

**Comment by jaustin** [comment]
@microbit-matt-hillsdon lancaster-university/microbit-v2-samples now has v0.2.9 and @pelikhan do you need to use this patch?

File: pxtarget.json [hunk 1]
```diff
@@ -142,14 +142,14 @@
...
+                    "name": "codal-microbit-v2",
+                    "url": "https://github.com/lancaster-university/codal-microbit-v2",
+                    "branch": "v0.2.9",
                     "type": "git"
                 },
                 "codalBinary": "MICROBIT",
-                "githubCorePackage": "microbit-foundation/codal",
-                "gittag": "pxt-wip-build-v0.0.1",
+                "githubCorePackage": "lancaster-university/microbit-v2-samples",
+                "gittag": "master",
```
Comment (pelikhan) [inline pxtarget.json:152]: we don't work of master

**Review by pelikhan — CHANGES_REQUESTED** [review]
Please add a tag.

File: pxtarget.json (hunk 1 above)
Comment (pelikhan) [reply pxtarget.json:152]: what is this repo? lancaster-university/microbit-v2-samples

**Comment by jaustin** [comment]
@pelikhan as per Matt's list of TODO
> - [ ] Missing a tag for microbit-v2-samples so that's using master as the gittag. We've asked Joe to tag it.

Joe's done the tag now.

File: pxtarget.json (hunk 1 above)
Comment (pelikhan) [reply pxtarget.json:152]: @microbit-matt-hillsdon Joe just did a v0.2.9 tag

Comment (microbit-matt-hillsdon) [reply pxtarget.json]: Pushed 3b387a2751d2232038961b2be179976f0df30ea6 with the tag though it doesn't build. Not sure why.

Tag is definitely there: https://github.com/lancaster-university/microbit-v2-samples/releases/tag/v0.2.9

Any idea what I did wrong?

Comment (microbit-matt-hillsdon) [reply pxtarget.json]: > what is this repo? lancaster-university/microbit-v2-samples

@pelikhan this is the new public repo for codal for v2.

Still puzzled by the build issue. Any chance the build doesn't fetch tags?

```
Cloning into '/home/build/prjclone'...
warning: Could not find remote branch v0.2.9 to clone.
fatal: Remote branch v0.2.9 not found in upstream origin
```

**Comment by pelikhan** [comment]
@microbit-matt-hillsdon @mmoskal sorted out the CODAL binding issues. Can you pull master again?

**Comment by microbit-matt-hillsdon** [comment]
> @microbit-matt-hillsdon @mmoskal sorted out the CODAL binding issues. Can you pull master again?

@pelikhan great, thanks. Pushed a merge and the build looks to be past the C++ now (still in progress as I write). I've ticked the box that means maintainers can push to this branch so if that's useful to make progress while we don't have timezone overlap then go for it.

File: pxtarget.json (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply pxtarget.json]: Build looks good now.

**Comment by microbit-matt-hillsdon** [comment]
@pelikhan what's the best approach to get the simulator working again for the volume blocks? I tried adding a pxtsim.music namespace with volume/setVolume (similar to other code in edgeconnector.ts) but that didn't seem to make any difference. Perhaps there's some other approach?

**Comment by pelikhan** [comment]
You want to add a exported "setVolume" function in the simulator part ( https://github.com/microsoft/pxt-microbit/blob/8ef4fc63b68568e3812c92fc7ed70e1719e6f849/sim/state/edgeconnector.ts#L104 ).

**Comment by pelikhan** [comment]
This should work
```
export function setVolume(volume: number): void {
        analogSetPitchVolume(volume);
}
```

**Comment by microbit-matt-hillsdon** [comment]
Thanks @pelikhan, the simulator works now. I had it roughly right but had typo'd pxsim as pxtsim (needs the music namespace).

I've removed the todo about testing the build as that's already gone in in #3449.

**Comment by pelikhan** [comment]
@microbit-matt-hillsdon did you test this on hardware v1/v2?

**Review by pelikhan — APPROVED** [review]

**Comment by microbit-matt-hillsdon** [comment]
> @microbit-matt-hillsdon did you test this on hardware v1/v2?

@pelikhan yes, it's working on both. There's a difference in volume which I think it makes sense to look at separately (not yet clear where best to fix). I'll follow up on that.

**Pull request merged**